EventCatalogAPI/Controllers/EventItemsController.cs

```diff
@@ -63,5 +68,34 @@ public async Task<IActionResult> Dates()
             return Ok(events);
         }
 
+        //List of Event Categories
+        [HttpGet]
+        [Route("[action]")]
+
+        public async Task<IActionResult> CategoryLists()
+        {
+            var events = await _context.EventCategory.ToListAsync();
+            return Ok(events);
+        }
+
+
+        //Sort Event by Category 
+        [HttpGet]
+        [Route("[action]")]
+        public async Task<IActionResult> EventCategories(
+            [FromQuery] int pageIndex = 0,
+            [FromQuery] int pageSize = 4)
```

Comment:
We will need to accept an input category FromQuery - something like this: "[FromQuery] string eventCategory".